### PR TITLE
[DSS-415] Checkbox Refactor (Styles, Event, Tests)

### PR DESCRIPTION
**Pull request by pixelflips**
# Description

Minor refactoring to checkbox component based on new agreed-upon code conventions and small issues found. 
No visual or functional changes are expected.

- CSS prop naming clean up
- JSX lint warning fix
  
Screenshot 2023-06-07 at 3 18 42 PM

- Incorrect ID naming in the e2e test

Fixes [DSS-415](https://kajabi.atlassian.net/browse/DSS-415)

## Type of change

- [X] Refactor (Code changes that neither fix a bug nor adds a feature)

# How Has This Been Tested?

Unit & End-to-end tests have been added for the component.
To test run the `yarn nx test core` command and verify the passing results.

- [x] unit tests
- [x] e2e tests
- [x] tested manually

**Test Configuration**:

- os: macOS 13.4
- browsers: Chrome(latest), Firefox(latest), Safari(latest)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

[DSS-415]: https://kajabi.atlassian.net/browse/DSS-415?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ju-Skinner — APPROVED** [review]

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
This is blocked until after PR #68 is merged.

**Comment by pixelflips** [comment]
> This is blocked until after PR #68 is merged.

@ju-Skinner Makes sense. Depending on the conflicts I may just redo this one in another PR. Let's see how it goes first. Thanks for the heads up.

**Comment by ju-Skinner** [comment]
Replaced by #72

**Pull request closed without merging**
